---
title: "Time-dependent Effects of SNP on the Alzheimer's Disease Longitudinal Measurements"
author: "Hao Wu"
date: "10/28/2022"
output:
  slidy_presentation: 
    duration: 30
    footer: "Copyright (c) 2022, Hao Wu, UConn"
bibliography: ref.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library('tidyverse')
```


# Abstract

---

Alzheimer's disease (AD) is a serious neurodegenerative condition that affects millions of individuals across the world and the most common cause of dementia. As the average age of individuals in the world increases, the prevalence of AD will continue to grow. The Alzheimer’s Disease Neuroimaging Initiative (ADNI) unites researchers with study data as they work to define the progression of Alzheimer’s disease. ADNI researchers collect, validate, and utilize data, including MRI and PET images, genetics, cognitive tests, Cerebrospinal Fluid (CSF) and blood biomarkers as predictors of the disease. The genetic data from the Alzheimer's Disease Neuroimaging Initiative (ADNI) have been crucial in advancing the understanding of Alzheimer's disease (AD). 

The core element of ADNI genetic data is single-nucleotide polymorphism (SNP), which is always coded by 3 values: 0, 1 and 2.  Considering the fact that the value of SNP does not change over time, we apply a screening procedure for varying coefficient models with ultrahigh dimensional longitudinal predictor variables, to find the time-dependent effects of SNPs in response variable Mini Mental State Examination (MMSE), which is a tool that can be used to systematically and thoroughly assess mental status.




# Introduction


## DNA bases

DNA base pair is the molecules called nucleotides, on opposite strands of the DNA double helix, that form chemical bonds with one another. These chemical bonds act like rungs in a ladder and help hold the two strands of DNA together.

There are four nucleotides, or bases, in DNA: adenine (A), cytosine (C), guanine (G), and thymine (T). These bases form specific pairs ( A with T, and G with C).

Number of DNA bases of humanbeing

The human genome contains approximately __3 billion__ of these base pairs, which reside in the 23 pairs of chromosomes within the nucleus of all our cells.


```{r, out.width='50%', fig.align='center'}
knitr::include_graphics('./Figure/Intro_1.jfif')
```


## SNP

In genetics, a single-nucleotide polymorphism ( SNP, pronounced "snips") is a germline substitution of a single nucleotide at a specific position in the genome.

For example, at a specific base position in the human genome, the G nucleotide may appear in most individuals, but in a minority of individuals, the position is occupied by an A. This means that there is a SNP at this specific position, and the two possible nucleotide variations - G or A - are said to be the alleles for this specific position.

SNPs occur normally throughout a person's DNA. They occur almost once in every 1,000 nucleotides on average, which means there are roughly 4 to 5 million SNPs in a person's genome.


## One SNP example

The upper DNA molecule differs from the lower DNA molecule at a single base-pair location (a G/A polymorphism)

```{r, out.width='50%', fig.align='center'}
knitr::include_graphics('./Figure/Intro_2.jfif')
```


## Genome-wide association study (GWAS)
One of main contributions of SNPs in clinical research is genome-wide association study (GWAS). Gnome-wide genetic data can be generated by multiple technologies, including SNP array and whole genome sequencing.

GWAS has been commonly used in identifying SNPs associated with diseases or clinical phenotypes or traits. Since GWAS is a genome-wide assessment, a large sample site is required to obtain sufficient statistical power to detect all possible associations.



# Data

## Patients' information
In the ADNI1 dataset, we totally had 1843 patients. However, there were 1250 (818 + 432) patients who had genetic data, which lies on the folders "ADNI1","ADNIGO2" and "ADNI3" under the directory "/Genetic/GWAS/". The total size of the dataset is around 50GB (all .zip file).

- 819 patients and 5122 longitudinal observations.

```{r}
## Original covariates
data_covariates <- read.csv("ADNI_Patient_Covariates.csv")
variable_list <- c("RID", "VISCODE", "EXAMDATE","AGE","PTGENDER","MMSE", "APOE4")
data_covariates <- data_covariates %>%
  filter(ORIGPROT == "ADNI1") %>%
  dplyr::select(variable_list) %>% 
  drop_na(MMSE)



## Transformation
data_patient_ADNI1 <- data_covariates %>%
  mutate(EXAMDATE = as.numeric(as.Date(EXAMDATE, "%Y-%m-%d"))) %>%
  mutate(EXAMDATE = EXAMDATE - min(EXAMDATE)) %>%
  mutate(AGENEW   = AGE - min(AGE)) %>%
  mutate(GENDER   = if_else(PTGENDER == "Female",
                            true = 0,
                            false = 1))


data_patient_ADNI1 <- data_patient_ADNI1 %>%
  mutate(EXAMDATE2 = EXAMDATE/365) %>%
  group_by(RID) %>%
  mutate(AgeDiff   = EXAMDATE2 - min(EXAMDATE2)) %>%
  mutate(AGENEW    = AGENEW  + AgeDiff)


head(data_patient_ADNI1)
```


## One patient genetic dataset 

We picked one patient (RID:0413) as our example.

- SNPs (Rows)
There are 620901 SNPs (rows) for each patient.

- Variables (Columns)
There are 24 variables (columns) for each patient.

Index Variables Sample Index/ Sample ID / SNP Index / SNP Name / SNP / Chr(chromosome) / Position

Strands Allele1 - Top / Forward / AB, Allele2 - Top / Forward / AB

Call Scores GC Score / GT Score / Cluster Sep

Values Theta / R / X / Y / X Raw / Y Raw / B Allele Freq / Log R Ratio





We put the focus on following variables:  "Allele1AB", "Allele2AB", "SNPName", "SNP", "Chromosome", "Position".



- Allele1AB means Allele 1 in A/B notation
- Allele2AB means Allele 2 in A/B notation
- Chr includes 26 typeS: 1-22, X, Y, XY, MT
- Position is the chromosomal position of the SNP
- SNP includes 8 types, "[T/C]", "[A/G]", "[C/G]", "[T/A]", "[G/C]", "[T/G]", "[A/C]" and "[A/T]".




## How to set time points
- Treat the smallest age of patients (ADNI1) as 0,
- Treat the age at baseline of each patient as its original age, then add the date difference over 365 to the original age as its new age.



# Methods

---

```{r pressure}
plot(pressure)
```




# Acknowledgement

---

Data collection and sharing for this project was funded by the ADNI(National Institutes of Health Grant U01 AG024904) and DOD ADNI (Department of Defense award number W81XWH-12-2-0012). ADNI is funded by the National Institute on Aging, the National Institute of Biomedical Imaging and Bioengineering, and through generous contributions from the following: AbbVie, Alzheimer’s Association; Alzheimer’s Drug Discovery Foundation; Araclon Biotech; BioClinica, Inc.; Biogen; Bristol-Myers Squibb Company; CereSpir, Inc.; Cogstate; Eisai Inc.; Elan Pharmaceuticals, Inc.; Eli Lilly and Company; EuroImmun; F. Hoffmann-La Roche Ltd and its affiliated company Genentech, Inc.; Fujirebio; GE Healthcare; IXICO Ltd.; Janssen Alzheimer Immunotherapy Research \& Development, LLC.; Johnson \& Johnson Pharmaceutical Research \& Development LLC.; Lumosity; Lundbeck; Merck \& Co., Inc.; Meso Scale Diagnostics, LLC.; NeuroRx Research; Neurotrack Technologies; Novartis Pharmaceuticals Corporation; Pfizer Inc.; Piramal Imaging; Servier; Takeda Pharmaceutical Company; and Transition Therapeutics. The Canadian Institutes of Health Research is providing funds to support ADNI clinical sites in Canada. Private sector contributions are facilitated by the Foundation for the National Institutes of Health (www.fnih.org). The grantee organization is the Northern California Institute for Research and Education, and the study is coordinated by the Alzheimer’s Therapeutic Research Institute at the University of Southern California. ADNI data are disseminated by the Laboratory for Neuro Imaging at the University of Southern California. A special thanks to Danielle J Harvey for her support in helping us navigate the ADNI database.




# Reference

---

---
nocite: '@*'
---